caveclient/chunkedgraph.py

```diff
@@ -689,14 +797,41 @@ def remesh_level2_chunks(self, chunk_ids):
         r.raise_for_status()
 
     def get_operation_details(self, operation_ids: Iterable[int]):
-        """get the details of a list of operations
+        """Get the details of a list of operations.
 
-        Args:
-            operation_ids (Iterable[int]): list of operation IDss
+        Parameters
+        ----------
+        operation_ids: Iterable of int
+            List/array of operation IDs.
 
-        Returns:
-            dict: a dict of dictss of operation info, keys are operationids
-            values are a dictionary of operation info for the operation
+        Returns
+        -------
+        dict of str to dict
+            A dict of dicts of operation info, keys are operation IDs (as strings),
+            values are a dictionary of operation info for the operation. These
+            dictionaries contain the following keys:
+
+            "added_edges": list of list of int
+                List of edges added by this operation. Each edge is a list of two node
+                IDs (source and target).
+            "roots": list of int
+                List of root IDs that were created by this operation.
+            "sink_coords": list of list of int
+                List of sink coordinates for this operation. The sink is one of the
+                points placed by the user when specifying the operation. Each sink
+                coordinate is a list of three integers (x, y, z), corresponding to
+                spatial coordinates in segmentation voxel space.
+                # TODO make sure this is the correct specification
+            "source_coords": list of list of int
+                List of source coordinates for this operation. The source is one of the
+                points placed by the user when specifying the operation. Each source
+                coordinate is a list of three integers (x, y, z), corresponding to
+                spatial coordinates in segmentation voxel space.
+                # TODO make sure this is the correct specification
```

Comment:
can someone double check that A) this is the correct specification for what these fields mean, and B) "segmentation voxel space" is an appropriate descriptor? 

P.S. somewhere we should document the meanings of these potentially different spaces (voxel, segmentation voxel(?), nm, etc).

Reply:
The input source and sink coordinates are in nm space, neuroglancer converts to nm and chunkedgraph converts it back to voxel space because it is not guaranteed they both use the same resolution.

Other than this rest of the pr looks good.

Reply:
@akhileshh I don't think that is true that source/sink coords are in nm space, have a look at this example:

```python
import numpy as np

import caveclient as cc

client = cc.CAVEclient("minnie65_phase3_v1")

operation_id = 9408

out = client.chunkedgraph.get_operation_details([operation_id])[str(operation_id)]

sink_coords_point = out["sink_coords"][0]

print("Point (seg-voxel space) for sink_coords:", sink_coords_point)

modified_supervoxel = out["added_edges"][0][0]

l2_id = client.chunkedgraph.get_roots([modified_supervoxel], stop_layer=2)[0]

l2_point = client.l2cache.get_l2data(l2_id, attributes=["rep_coord_nm"])[str(l2_id)][
    "rep_coord_nm"
]
print("Point (nm) from level 2 cache:", l2_point)

print(
    "Point for sink_coords scaled by (8x8x40):",
    np.array(sink_coords_point) * np.array([8, 8, 40]),
)
```

yields:

```
Point (seg-voxel space) for sink_coords: [99032, 77136, 26391]
Point (nm) from level 2 cache: [792704.0, 615200.0, 1055280.0]
Point for sink_coords scaled by (8x8x40): [ 792256  617088 1055640]
```

Reply:
Point being that multiplying by 8x8x40 brings `sink_coords` close to where the L2 cache says that node is in nanometers. So I believe `sink_coords` is in "segmentation voxel" space, or whatever we want to call that.

Reply:
Oh you're right, my bad. I was confused by what was being documented here; I was referring to input to chunkedgraph from neuroglancer but indeed this endpoint returns the already converted coordinates in voxel space.